File: RobWork/example/cpp/ex-rotation3d.cpp
#include <rw/math/Rotation3D.hpp>

using rw::math::Rotation3D;

int main(int argc, char** argv) {
    Rotation3D<> rotd = Rotation3D<>(1,0,0,0,0,-1,0,1,0);
    Rotation3D<float> rotf = Rotation3D<float>(1,0,0,0,0,-1,0,1,0);

    std::cout << "Rotation double:" << std::endl << rotd << std::endl;
    std::cout << "Rotation float:" << std::endl << rotf << std::endl;
    std::cout << "Rotation inverse:" << std::endl << inverse(rotd) << std::endl;
    std::cout << "Identity:" << std::endl << rotd*inverse(rotd) << std::endl;

    return 0;
}

File: RobWork/example/python/ex-rotation3d.py
from sdurw import *

if __name__ == '__main__':
    rotd = Rotation3Dd(1,0,0,0,0,-1,0,1,0);
    rotf = Rotation3Df(1,0,0,0,0,-1,0,1,0);
    
    print("Rotation double:");
    print(str(rotd));
    print("Rotation float:");
    print(str(rotf));
    print("Rotation inverse:");
    print(str(inverse(rotd)));
    print("Identity:");
    print(str(rotd*inverse(rotd)));

    rotd[2,2] = 25;
    print("Corner = 25")
    print(str(rotd))

    if rotd[2,2] == 25:
        exit(0)
    else:
        exit(1)